sktime/decomposition/koopman.py

```diff
@@ -0,0 +1,575 @@
+        return self.cov.cov_0t
+
+    @property
+    def cov_tt(self) -> np.ndarray:
+        r""" Shortcut to :attr:`cov_tt <sktime.covariance.CovarianceModel.cov_tt>`. """
+        return self.cov.cov_tt
+
+    @property
+    def whitening_rank_0(self) -> int:
+        r""" Rank of the instantaneous whitening transformation :math:`C_{00}^{-1/2}`. """
+        return self._rank_0
+
+    @property
+    def whitening_rank_t(self) -> int:
+        r""" Rank of the time-lagged whitening transformation :math:`C_{tt}^{-1/2}`. """
+        return self._rank_t
+
+    @property
+    def epsilon(self) -> float:
+        r""" Singular value cutoff. """
+        return self._epsilon
+
+    @staticmethod
+    def _cumvar(singular_values) -> np.ndarray:
+        cumvar = np.cumsum(singular_values ** 2)
+        cumvar /= cumvar[-1]
+        return cumvar
+
+    @staticmethod
+    def effective_output_dimension(rank0, rankt, dim, singular_values) -> int:
+        r""" Computes effective output dimension. """
+        if dim is None or (isinstance(dim, float) and dim == 1.0):
+            return min(rank0, rankt)
+        if isinstance(dim, float):
+            return np.searchsorted(CovarianceKoopmanModel._cumvar(singular_values), dim) + 1
+        else:
+            return np.min([rank0, rankt, dim])
+
+    @property
+    def dim(self) -> Optional[Real]:
+        r""" Dimension attribute. Can either be int or float. In case of
+
+        * :code:`int` it evaluates it as the actual dimension, must be strictly greater 0,
+        * :code:`float` it evaluates it as percentage of the captured kinetic variance, i.e., must be in :math:`(0,1]`,
+        * :code:`None` all components are used.
+
+        :getter: yields the dimension
+        :setter: sets a new dimension
+        :type: int or float
+        """
+        return self._dim
+
+    @dim.setter
+    def dim(self, value: Optional[Real]):
+        if isinstance(value, Integral):
+            if value <= 0:
+                # first test against Integral as `isinstance(1, Real)` also evaluates to True
+                raise ValueError("VAMP: Invalid dimension parameter, if it is given in terms of the "
+                                 "dimension (integer), must be positive.")
+        elif isinstance(value, Real) and (value <= 0. or float(value) > 1.0):
+            raise ValueError("VAMP: Invalid dimension parameter, if it is given in terms of a floating point, "
+                             "can only be in the interval (0, 1].")
+        elif value is not None and not isinstance(value, (Integral, Real)):
+            raise ValueError("Invalid type for dimension, got {}".format(value))
+        self._dim = value
+        self._output_dimension = CovarianceKoopmanModel.effective_output_dimension(
+            self.whitening_rank_0, self.whitening_rank_t, self._dim, self.singular_values
+        )
+
+    @property
+    def cumulative_kinetic_variance(self) -> np.ndarray:
```

Comment:
how does this, or does it not, relate to `_cumvar`

Reply:
`_cumvar` is a static (ie stateless) private method which gets called by this property, agreed the name is perhaps not the best. on the other hand "cum" in the sense of cumulative is fairly standard (see `np.cumsum`) and so is `var` as a form of variance